src/pk_core.c

```diff
@@ -648,7 +648,37 @@ DEF(coreExit,
 // String functions.
 // -----------------
 
-// TODO: substring.
+DEF(coreStrSub,
+  "str_sub(str:string, pos:num, len:num) -> string\n"
+  "Returns a substring from a given string supplied. In addition, "
+  "the position and length of the substring are provided when this "
+  "function is called. For example: `str_sub(str, pos, len)`.") {
+  int c = 0;
+  int64_t len, pos;
+
+  String* str = NULL;
+  if (!validateArgString(vm, 1, &str)) return;
+  if (!validateInteger(vm, ARG(2), &pos, "Argument 2")) return;
+  if (!validateInteger(vm, ARG(3), &len, "Argument 3")) return;
+
+  if (str->length < 1)
+    RET_ERR(newString(vm, "Expected a string length more than 0."));
+
+  if (str->length < pos || str->length < len)
+    RET_ERR(newString(vm, "Index out of range."));
+
+  String* subStr = newStringLength(vm, str->data + pos, (uint32_t)len);
+
+  if (subStr == NULL)
+    RET_ERR(newString(vm, "Unable to allocate memory."));
```

Comment:
If there is an allocation failure, it should be handled at the VM's allocation function, otherwise we have to check if the allocation failed at every single place we're creating a new object. 

Remove these 2 lines.

Reply:
Also, don't use `camelCase` for local variables, instead use `snake_case` like `sub_str` for the convention. See #150 for a minimal naming conventions pocketlang's using.